src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/TransmissionStorage.cs

```diff
@@ -139,24 +140,32 @@ public virtual Transmission Dequeue()
                         return transmission;
                     }
                 }
-                catch (UnauthorizedAccessException)
+                catch (UnauthorizedAccessException uae)
                 {
+                    TelemetryChannelEventSource.Log.TransmissionStorageDequeueUnauthorizedAccessException(file?.Name ?? string.Empty, uae.ToString());
                     if (file == null)
                     {
-                        return null; // because the process does not have permission to access the folder
+                        return null; // Because the process does not have permission to access the folder.
                     }
 
                     if (lastInaccessibleFileName != file.Name)
                     {
                         lastInaccessibleFileName = file.Name;
-                        continue; // because another thread is loading this file right now
+                        continue; // Because another thread is loading this file right now.
+                    }
+                    else
+                    {
+                        // The same file has been inaccessible more than once.
+                        TelemetryChannelEventSource.Log.TransmissionStorageInaccessibleFile(file.Name);
                     }
 
-                    throw; // because the process does not have permission to modify this file
+                    throw; // Because the process does not have permission to modify this file.
                 }
-                catch (IOException)
+                catch (IOException ioe)
                 {
-                    continue; // because another thread already loaded this file
+                    TelemetryChannelEventSource.Log.TransmissionFailedToStoreWarning(file.Name, ioe.ToString());
+                    Thread.Sleep(random.Next(1, 100)); // Sleep for random time of 1 to 100 milliseconds to try to avoid future timing conflicts.
```

Comment:
I wonder - if we just return an empty transmission here instead of waiting - will the caller of this method wait before the scheduling of the next read? I know if was already working like this, but ultimately in the scenario we saw with the only one completely broken file - this task will never complete. Maybe we can release task and caller will wait before scheduling it again.